### PR TITLE
chore: update GitHub actions workflows

### PR DESCRIPTION
**Pull request by tylervz**
Update the GitHub actions workflows to use the latest version of multiple actions, including `actions/checkout`,  `actions/setup-java`, `actions/cache`, and `release-drafter/release-drafter`:

https://github.com/actions/checkout/tags
https://github.com/actions/setup-java/tags
https://github.com/actions/cache?tab=readme-ov-file#v4
https://github.com/release-drafter/release-drafter/tags

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -13,44 +13,54 @@ jobs:
...
+        with:
+          # Publish a Gradle build scan so you can see all the build logs,
+          # a complete task timeline, test outputs, and the resolved dependencies of your build.
+          build-scan-publish: true
+          build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
+          build-scan-terms-of-use-agree: "yes"
       - name: Run Build
         id: build
-        uses: gradle/gradle-build-action@v2
+        run: ./gradlew build
```
Comment (tylervz) [inline .github/workflows/gradle.yml:29]: The `gradle/actions/setup-gradle` action replaces `gradle/gradle-build-action`.
Reference:
https://github.com/gradle/actions/tree/af6e5767241073b8ab20cc8cf7e95ec63fadc160/setup-gradle#the-setup-gradle-action

File: .github/workflows/release-notes.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
         id: extract_branch
         run: echo "value=${GITHUB_REF:11}" >> $GITHUB_OUTPUT
       # If it has release drafter:
-      - uses: release-drafter/release-drafter@v5
+      - uses: release-drafter/release-drafter@v6
```
Comment (tylervz) [inline .github/workflows/release-notes.yml:27]: There are no breaking changes in this release:
https://github.com/release-drafter/release-drafter/releases/tag/v6.0.0

**Comment by tylervz** [comment]
I did not upgrade the ["Build and install groovy (no docs)" step in `groovy-joint-workflow.yml`](https://github.com/grails/grails-data-mapping/pull/1806/files#diff-cb75eba9bba0d6665705e7a07feb9286848cc414d3bcd32aadcbe925c5d96c3cL97-R111), because I did not see a suitable replacement using `gradle/actions/setup-gradle@v3`.

If I replaced it with `gradle/actions/setup-gradle@v3` like so:
```yaml
- name: Setup Gradle
uses: gradle/actions/setup-gradle@v3
with:
  build-scan-publish: true
  build-scan-terms-of-use-url: "https://gradle.com/terms-of-service"
  build-scan-terms-of-use-agree: "yes"
# Use the Gradle wrapper in the ../groovy directory
- name: Build and install groovy (no docs)
  run: |
    cd ../groovy
    ./gradlew install \
      -x groovydoc \
      -x javadoc \
      -x javadocAll \
      -x groovydocAll \
      -x asciidoc \
      -x docGDK
  env:
    GRADLE_SCANS_ACCEPT: yes
    GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
```
Then the `gradle/actions/setup-gradle@v3` step in the example above is useless since it is configuring Gradle in the root directory for optimal execution rather than the Gradle wrapper in `../groovy` that is actually going to get used.

I do not see any inputs/parameters for `gradle/actions/setup-gradle@v3` similar to `build-root-directory` to have it configure a Gradle wrapper in a directory other than the root directory.
https://github.com/gradle/actions/blob/main/setup-gradle/action.yml

Update: perhaps we should update the step to be the same as it is in `grails-core`?
https://github.com/grails/grails-core/blob/6.2.x/.github/workflows/groovy-joint-workflow.yml#L103-L119

**Comment by jamesfredley** [comment]
@tylervz after https://github.com/grails/grails-data-mapping/pull/1849 has been merged, can you rebase this against 9.0.x?

**Comment by tylervz** [comment]
@jamesfredley yes, I can do that :)

**Comment by tylervz** [comment]
I've rebased my branch on `9.0.x`. I'm going to close this pull request and open a new one requesting to be merged into `9.0.x`.

**Pull request closed without merging**